src/pru-motion-queue.cc

```diff
@@ -107,15 +111,23 @@ static void DumpMotionSegment(volatile const struct MotionSegment *e) {
 }
 #endif
 
+const struct QueueStatus PRUMotionQueue::get_status(){
+    //Snapshot the status register
+    struct QueueStatus status_register;
+    memcpy(&status_register, (struct QueueStatus *)&pru_data_->status, sizeof(struct QueueStatus));
```

Comment:
Given that you anyway memcpy() things, mabye GetStatus() should actually get a QueueStatus* paramter that is filled in.